### PR TITLE
chore: bump Nx to v17

### PR DESCRIPTION
**Pull request by tido64**
### Description

Bump Nx to [v17](https://blog.nrwl.io/nx-17-0-has-landed-1dad19176ff3)

### Test plan

n/a

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,7 +56,7 @@
     "typescript": "^5.0.0"
   },
   "engines": {
-    "node": ">=14.18"
+    "node": ">=16.13"
```
Comment (kelset) [inline package.json:59]: do we have some internal customer still on 16? would be nice to keep it at LTS level but I guess we have a bunch of people still at 18

**Review by kelset — APPROVED** [review]
overall look good

File: package.json (hunk 1 above)
Comment (kelset) [reply package.json:59]: right, so can't we bump it to at least 18? is there some CI VM or something that still runs on 16 so we need to keep it around?

Comment (tido64) [reply package.json:59]: Yup, we run Node versions 16, 18, 20 on CI:

https://github.com/microsoft/rnx-kit/blob/7aa6708cdf3bb1a71f4fd7ea602ee57731883f63/.github/workflows/pr.yml#L50-L54

Comment (kelset) [reply package.json:59]: got it - go ahead with merging 👍

**Pull request merged**